packages/fast-check/src/arbitrary/_internals/helpers/TokenizeString.ts

```diff
@@ -0,0 +1,56 @@
+import type { Arbitrary } from '../../../check/arbitrary/definition/Arbitrary';
+import { safePop, safePush, safeSubstring } from '../../../utils/globals';
+
+/**
+ * Split a string into valid tokens of patternsArb
+ * @internal
+ */
+export function tokenizeString(patternsArb: Arbitrary<string>, value: string): string[] | undefined {
```

Comment:
We might want to put an early exit if we are going too far in the number of chunks while we know for sure that in the current context there is no need to go that far. But it could be an optim for later. It's probably way less critical than the optim we are going to unlock now.